app/components/machine/driver-hwclouds/component.js

```diff
@@ -0,0 +1,377 @@
+import Ember from 'ember';
+import Driver from 'ui/mixins/driver';
+
+const VOLUME_TYPES = [{
+  name: 'SAS',
+}, {
+  name: 'SATA',
+}, {
+  name: 'SSD',
+}, ];
+
+const URLS = {
+  ecs:  'https://ecs.eu-de.otc.t-systems.com',
+};
+
+export default Ember.Component.extend(Driver, {
+  driverName:       'hwcloudsConfig',
+  catalogUrls:      null,
+  step:             1,
+  intl:             Ember.inject.service(),
+  isStep1:          Ember.computed.equal('step', 1),
+  isStep2:          Ember.computed.equal('step', 2),
+  isStep3:          Ember.computed.equal('step', 3),
+  isStep4:          Ember.computed.equal('step', 4),
+  isGteStep2:       Ember.computed.gte('step', 2),
+  isGteStep3:       Ember.computed.gte('step', 3),
+  isGteStep4:       Ember.computed.gte('step', 4),
+  volumeTypes:      VOLUME_TYPES,
+  itemsLoading:     false,
+  flavors:          null,
+  images:           null,
+  osAvailabilityZone:   null,
+  subnet:           null,
+
+  init: function() {
+    this._super(...arguments);
+    if (!this.get('clonedModel')) {
+      this.initModel();
+    }
+  },
+
+
+  initModel: function() {
+    var config = this.get('store').createRecord({
+      type:             'hwcloudsConfig',
+      serviceEndpoint:  URLS.ecs,
+      region:           'eu-de',
+      sshUser:          'linux',
+    });
+
+    var type = 'host';
+
+    if (!this.get('useHost')) {
+      type = 'machine';
+    }
+
+    this.set('model', this.get('store').createRecord({
+      type:              type,
+      'hwcloudsConfig':  config,
+    }));
+  },
+
+  actions: {
+    authorizeCreds: function() {
+      this.set('step', 2);
+      this.getZones().then(() => {
+        this.getNetworks().then(() => {
+        });
+      });
+    },
+    goToStep3: function() {
+      this.set('step', 3);
+      this.getSubnets().then(() => {
+        this.getSecurityGroups();
+      });
+    },
+    goToStep4: function() {
+      this.set('step', 4);
+      this.getFlavors().then(() => {
+        this.getImage();
+      });
+    },
+
+  },
+
+
+  getURL: function(type, version, projectId, endpoint, qp) {
```

Comment:
projectId is always `this.get('model.hwcloudsConfig.tenantId')`, doesn't seem like it needs to be an argument